fastplotlib/graphics/mesh.py

```diff
@@ -0,0 +1,473 @@
+from typing import Sequence, Any, Literal
+
+import numpy as np
+
+import pygfx
+
+from ._positions_base import Graphic
+from .features import (
+    MeshVertexPositions,
+    MeshIndices,
+    MeshCmap,
+    SurfaceData,
+    surface_data_to_mesh,
+    VertexColors,
+    UniformColor,
+    resolve_cmap_mesh,
+    VolumeSlicePlane,
+    PolygonData,
+    triangulate_polygon,
+)
+
+
+class MeshGraphic(Graphic):
+    _features = {
+        "positions": MeshVertexPositions,
+        "indices": MeshIndices,
+        "colors": (VertexColors, UniformColor),
+        "cmap": MeshCmap,
```

Comment:
@almarklein your original question:

> Was the `mapcoords` feature removed intentionally? I can see it being useful. Maybe I don't properly understand features vs normal properties.

"Graphic Features" are evented properties. You can add event handlers to them to capture when they change.

The `BufferManager` is also a graphic feature, but allows numpy-like slicing and setting and automatically calls `update_range()` on the `pygfx.Buffer` object. 

Since `mapcoords` are usually changed in conjunction with a change in positions or cmap, are there use cases to change the `mapcoords` by itself without changing other things? If so then it should be a feature too.

Reply:
Ah I understand it now and how it's the same as cmap_transform but multi dimensional. One easy usecase is heat diffusion in 3d objects. I added it to #959 so we can do it later. Thanks for the context and ideas, I rarely do things with meshes! :smile: